src/packages/core/src/types.ts

```diff
@@ -199,3 +199,62 @@ export interface BackendProviderConfig {
 		root: boolean;
 	};
 }
+
+export type Constructor<T extends object, Arguments extends unknown[] = any[]> = new (
+	...arguments_: Arguments
+) => T;
+
+export type ClassType<T extends object = object, Arguments extends unknown[] = any[]> = Constructor<
+	T,
+	Arguments
+> & {
+	prototype: T;
+};
+
+// We want TypeValues to be able to just be generic Functions as well.
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type TypeValue = ClassType | GraphQLScalarType | Function | object | symbol;
+
+export type GetTypeFunction = (type?: void) => TypeValue;
+
+export type Complexity = ComplexityEstimator | number;
+
+export interface FieldMetadata<G, D> {
+	adminUIOptions?: {
+		hideInTable?: boolean;
+		hideInFilterBar?: boolean;
+		readonly?: boolean;
+	};
+	apiOptions?: {
+		excludeFromBuiltInWriteOperations?: boolean;
+	};
+	target: G;
+	name: string;
+	getType: GetTypeFunction;
+	relationshipInfo?: {
+		relatedField?: string;
+		id?: string | ((dataEntity: D) => string | number | undefined);
+	};
+	description?: string;
+	deprecationReason?: string;
+	complexity?: Complexity;
+	defaultValue?: any;
+
+	// This marks the field as read only in both the API and the admin UI.
+	// This will supersede any other read only settings.
+	readonly?: boolean;
+
+	nullable?: boolean | 'items' | 'itemsAndList';
+	excludeFromFilterType?: boolean;
+	hideInAdminUI?: boolean;
```

Comment:
```suggestion
```
Users can use hideInTable and hideInFilterBar together if they wish.

Reply:
I cant see any refs to hideInAdminUI so just removed.